### PR TITLE
[Silabs]Fix efr32 test driver event loop, run gtests, enable more test on efr…

### LINKED ISSUE #10447: EFR32 Unit Test crashes
#### Problem
EFR32 Unit tests crashes while running:
`python -m nl_test_runner.nl_test_runner -d /dev/ttyACM0 -f out/debug/chip-efr32-device_tests.s37 -o out.log`

```
...
<detail> [IN] New secure session created for device 0x0000000006A11E3D, key 2!!
<detail> [IN] New secure session created for device 0x000000000001E306, key 1!!
<info  > [IN] Prepared encrypted message 0x20020518 to 0x0000000006A11E3D of type 0xhx and protocolId (1, 0) on exchange 4� with
 MessageCounter:105.
<info  > [IN] Sending encrypted msg 0x20020518 with MessageCounter:1 to 0x0000000006A11E3D at monotonic time: ld msec
<info  > [-] 
FREERTOS ASSERT ( (pxLink->xBlockSize & xBlockAllocatedBit) != 0 )

<info  > [-] 
FREERTOS ASSERT ( ( portNVIC_INT_CTRL_REG & portVECTACTIVE_MASK ) == 0 )
```

### LINKED ISSUE #33373: [Test Failed] EFR32 TestExchange.cpp asserts
### Test issue(s)

While Running the test driver on efr32 hardware,
the device asserts in the test `TestExchange.cpp ; CheckBasicMessageRounTrip` in the function 

```
CHIP_ERROR ExchangeContext::SendMessage(Protocols::Id protocolId, uint8_t msgType, PacketBufferHandle && msgBuf,
                                        const SendFlags & sendFlags)
```
line 
`VerifyOrReturnError(mExchangeMgr != nullptr, CHIP_ERROR_INTERNAL);`

Before the assert, logs show events failing to get posted to the chip event queue because the queue is full. 
```
00> [00:00:13.921][error ][DL] Failed to post event to CHIP Platform event queue
00> [00:00:13.929][error ][DL] Failed to post event to CHIP Platform event queue
00> [00:00:13.936][info  ][-] src/messaging/tests/TestExchange.cpp:166: assertion failed: "delegate2.mReceivedMessageCount == 1"
```

### Platform

efr32

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by jmartinez-silabs**
fixes #10447
fixes #33373

- gtest were not run by the runner. add Call `chip::test::RunAllTests();`
- event loop processing in tests changed and was not working on efr32 anymore.  Add `CHIP_SYSTEM_CONFIG_USE_OPEN_THREAD_ENDPOINT` conditions in TestInetCommonOptions.cpp and TestInetCommonPosix.cpp to enable it.
- re-enable MessagingLayerTests tests.

### CODE REVIEW

File: src/app/icd/server/tests/TestICDManager.cpp [hunk 1]
```diff
@@ -1061,10 +1061,14 @@ TEST_F(TestICDManager, TestICDStateObserverOnTransitionToIdleModeEqualActiveMode
 
     // Expire IdleMode timer
     AdvanceClockAndRunEventLoop(1_s);
-    EXPECT_FALSE(mICDStateObserver.mOnTransitionToIdleCalled);
+    // In this scenario, The ICD state machine kicked a OnTransitionToIdle timer with a duration of 0 seconds.
+    // The freeRTOS systemlayer timer calls a 0s timer's callback instantly while on posix it take and 1 addition event loop.
+    // Thefore, the expect result diverges here based on the systemlayer implementation. Skip this check.
+    // EXPECT_FALSE(mICDStateObserver.mOnTransitionToIdleCalled);
```
Comment (arkq) [inline src/app/icd/server/tests/TestICDManager.cpp:1067]: Disabling this does not look like EFR32 specific. Maybe it should be done in a separate PR?

**Review by arkq — APPROVED** [review]

File: src/app/icd/server/tests/TestICDManager.cpp (hunk 1 above)
Comment (jmartinez-silabs) [reply src/app/icd/server/tests/TestICDManager.cpp:1067]: This check fails test fails. I assume it is not only silabs, and applies to other platform using the free RTOS-based system layer timer implementation.

I'd also add that the check commented here is a false assumption and was working only due to how the event loop works in test for posix implementation

Comment (mkardous-silabs) [reply src/app/icd/server/tests/TestICDManager.cpp:1067]: @jmartinez-silabs can you open an issue and add the icd label?

**Review by mkardous-silabs — APPROVED** [review]

File: src/app/icd/server/tests/TestICDManager.cpp (hunk 1 above)
Comment (jmartinez-silabs) [reply src/app/icd/server/tests/TestICDManager.cpp:1067]: I created https://github.com/project-chip/connectedhomeip/issues/33441

**Pull request merged**